### PR TITLE
Updated requirements.cli.txt to resolve bug #315

### PR DESCRIPTION
**Pull request by hvaria**
Switching to opencv-python-headless addresses the libGL.so.1 error encountered in headless environments by eliminating the dependency on GUI-related libraries that OpenCV's standard version requires. The headless variant of OpenCV is specifically compiled without these graphical components, making it ideal for server or Docker environments where graphical interfaces are absent. This solution directly tackles the issue by removing the need for the missing libraries, rather than attempting to add unnecessary graphical support to a headless setup. It ensures that applications utilizing OpenCV for image processing or computer vision tasks can operate smoothly without encountering errors related to graphics libraries, thus maintaining functionality while adhering to the constraints of the deployment environment.

### CODE REVIEW

**Comment by PawelPeczek-Roboflow** [comment]
Hi, thanks for contribution.
I responded on the topic of headless-opencv [here](https://github.com/roboflow/inference/issues/315#issuecomment-2003765836) and I am not sure if that change solves the problem, but I like other changes in the PR. Could we merge other changes first and the find out best solution for OpenCV as separated topic?

**Review by hvaria — COMMENTED** [review]
Enhanced error handling and messaging in server.py to provide users with clearer guidance when Docker is not running but is required for operation. Previously, the CLI would attempt to perform Docker-dependent tasks without first verifying Docker's status, leading to long and complex error messages that were difficult for users to diagnose and resolve. With these changes, the application now checks Docker's status upfront and provides a concise, informative error message if Docker is not running, guiding the user to start Docker to proceed.

**Comment by hvaria** [comment]
> Hi, thanks for contribution. I responded on the topic of headless-opencv [here](https://github.com/roboflow/inference/issues/315#issuecomment-2003765836) and I am not sure if that change solves the problem, but I like other changes in the PR. Could we merge other changes first and the find out best solution for OpenCV as separated topic?

I have updated my commits

**Comment by hvaria** [comment]
Earlier, the script threw long Docker error messages because it directly passed on the raw exceptions from Docker operations without filtering or simplifying them. When Docker wasn't running, the system tried to perform actions that required Docker, leading to complex error outputs from the underlying Docker library. These outputs were designed for developers, full of technical details, but not very helpful for everyday users trying to figure out a simple next step like starting Docker. My changes address this by catching these conditions early and providing a clear, simple message instead of letting those raw, detailed errors reach the user.

**Comment by PawelPeczek-Roboflow** [comment]
ok, maybe I am missing something, but I have a feeling that this is no longer the case.

This is the output from `main`:
```
❯ python -m inference_cli.main server status
Checking status of inference server.
Command failed. Cause: Error connecting to Docker daemon. Is docker installed and running? See https://www.docker.com/get-started/ for installation instructions.
```

**Comment by PawelPeczek-Roboflow** [comment]
do you have something else on your end?

**Comment by hvaria** [comment]
Nope, That's It.

**Comment by PawelPeczek-Roboflow** [comment]
We've decided to have this exact message as people were asking us about docker installation.

Do you find this message unclear? Your is `Docker is required but not running. Please start Docker to proceed with Docker-based inference.` - which seems to me similar - and I do think that it is important to provide link for Docker get-started page.

File: inference_cli/server.py [hunk 1]
```diff
@@ -59,40 +59,60 @@ def start(
         ),
     ] = None,
 ) -> None:
+
     try:
         ensure_docker_is_running()
+    except Exception as docker_error:
+        typer.echo("Docker is required but not running. Please start Docker to proceed with Docker-based inference.")
```
Comment (PawelPeczek-Roboflow) [inline inference_cli/server.py:66]: I would pust error as is here, without additional description

File: inference_cli/server.py [hunk 2]
```diff
@@ -59,40 +59,60 @@ def start(
...
 
 
 @server_app.command()
 def status() -> None:
-    typer.echo("Checking status of inference server.")
+    typer.echo("Checking status of the inference server.")
     try:
         ensure_docker_is_running()
+    except Exception as docker_error:
+        typer.echo("Docker is required to check the status but is not running. Please start Docker and try again.")
```
Comment (PawelPeczek-Roboflow) [inline inference_cli/server.py:91]: [1]

File: inference_cli/server.py [hunk 3]
```diff
@@ -59,40 +59,60 @@ def start(
...
 
 
 @server_app.command()
 def stop() -> None:
-    typer.echo("Terminating running inference containers")
+    typer.echo("Terminating running inference containers.")
     try:
         ensure_docker_is_running()
+    except Exception as docker_error:
+        typer.echo("Docker is required to stop containers but is not running. Please start Docker and try again.")
```
Comment (PawelPeczek-Roboflow) [inline inference_cli/server.py:108]: [1]

**Comment by PawelPeczek-Roboflow** [comment]
added comments. Once resolved I will merge

**Comment by hvaria** [comment]
Do you want me to resolve something or is it resolving something from your end before merging?

**Comment by PawelPeczek-Roboflow** [comment]
please resolve if you can

**Comment by hvaria** [comment]
@PawelPeczek-Roboflow , I removed the additional statements, as you mentioned.

**Comment by PawelPeczek-Roboflow** [comment]
what I meant here: `[1] I would post error as is here, without additional description`
was:
`typer.echo(f"{error}")`
not getting rid of error message

**Comment by PawelPeczek-Roboflow** [comment]
Thanks for the contribution. Will merge when CI completes

**Pull request merged**